src/api/app/models/bs_request_action.rb

```diff
@@ -900,6 +891,12 @@ def set_target_associations
     self.target_project_object = Project.find_by_name(target_project)
   end
 
+  def set_sourceupdate_default(user)
+    return unless action_type.in?([:submit, :maintenance_incident])
+    return if sourceupdate
+    update(sourceupdate: 'cleanup') if target_project && user.branch_project_name(target_project) == source_project
```

Comment:
Might be a matter of taste... I would combine the two if conditions

Reply:
fine with me